### ISSUE TITLE
Add reactions on updates, conversations and expenses

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3059]
# User story 
As part of the latest designs for conversations and expenses we introduced the ability to add reactions on them, quite similar to the ones you have on Github for issues and pull requests.

![image](https://user-images.githubusercontent.com/1556356/79373113-898f3400-7f56-11ea-83d6-fc0fe2188ebd.png)

## Best solution for this problem
See Conversations design: https://www.figma.com/file/e71tBo0Sr8J7R5n6iMkqI42d/%5BOC.com%5D-Web-%2F-04-Collectives-%E2%86%92-Create?node-id=7408%3A0

# MVP

The MVP should be about adding reactions on comment, that's the easiest to do and arguably the most important one because it will allow to add reactions on comments made on conversations, expenses (and updates once we will have #3053).

## API

- [x] Create a `CommentReactions` table
  - `UserId`
  - `CommentId`
  - `emoji`
  - add a unique index on these three columns (1 reaction type by user/comment)
- [x] Add the matching `CommentReaction` model
  - [x] `reaction` should be limited to a given set of pre-defined reactions
  - [x] `static addReaction(user, comment, reaction)`
  - [x] `removeReaction()`
- [x] Add a `addCommentReaction` mutation
- [x] Add a `removeCommentReaction` mutation
- [x] Create a GraphQL loader to load a map of the reactions for each comment. The map should look like this:
```es6
{
  '👍️': 10,
  '❤️': 7,
}
```
- [x] On GQLV2, add: `Comment.reactions` to fetch this reactions map

This won't provide the list of who reacted, but the cost of doing that being higher let's start simple and extend if needed.

## Frontend

- [x] Show reactions on comments according to the design on Figma

**Comment by Betree** [comment]
Updated with specs :rocket:

**Comment by SudharakaP** [comment]
If no one is already working on this; I'd like to take this issue. :smile:

**Comment by Betree** [comment]
@SudharakaP Sure! This one is more complex and the specs may need to be completed along the way. Feel free to ask for additional assistance if needed.

**Comment by SudharakaP** [comment]
> @SudharakaP Sure! This one is more complex and the specs may need to be completed along the way. Feel free to ask for additional assistance if needed.

Sure, thanks a bunch. Will let you know along the way. :smile:

**Comment by SudharakaP** [comment]
@Betree : Sorry that this took so long; I was busy with lots of stuff these days. :cry: :cry: 

But on a positive note, I've added basic support for addition of comments and both PRs are ready for review. Please let me know if there's anything further that needs to be done. Will try to get this to completion soon. :smile:

**Comment by SudharakaP** [comment]
Also a small demo of emoji addition; 

![Peek 2020-06-08 12-12](https://user-images.githubusercontent.com/12435965/84070812-92621c00-a981-11ea-863e-63beb09dffcc.gif)

**Comment by SudharakaP** [comment]
Also I just noticed that we are using `react-bootstrap` version 0.32.4. Is there a reason for this? :thinking: Can I upgrade to latest so that we can leverage a bit more nicer styles for the popup for emojis. :smile:

**Comment by Betree** [comment]
> Sorry that this took so long; I was busy with lots of stuff these days.

No worry, thanks for keeping us posted :slightly_smiling_face: 

> Also I just noticed that we are using react-bootstrap version 0.32.4. Is there a reason for this?

As per our [code guidelines](https://docs.opencollective.com/help/developers/best-practices-and-code-guidelines), we're getting rid of `Bootstrap`. Please don't use it for new developments.

For new features we usually rely on styled-components and our internal design system, for which must components are documented on https://styleguide.opencollective.com/

**Comment by SudharakaP** [comment]
@Betree : Thanks for the feedback. So I've used `react-popper` as you suggested without using any `react-bootstrap`. Herewith I have attached a screencast. Feel free to let me know if you have any further suggestions. I've tried to keep the emoji popover and stuff minimal in design as possible as I thought it might change in the future anyways (and the figma design doesn't seem to have an open popover). :smile: 

![Peek 2020-06-09 17-37](https://user-images.githubusercontent.com/12435965/84214493-5d84c080-aa78-11ea-92db-347afc4ee432.gif)
